core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -132,6 +132,23 @@ class StorageLevel private(
 
   override def hashCode(): Int = toInt * 41 + replication
 
+  /** Name of the storage level if it is predefined or [[None]] otherwise. */
+  def name: Option[String] = this match {
```

Comment:
`fromString` below has basically the opposite of this. How about storing the mapping in a `Seq[(StorageLevel, String)]` and using that in both methods? e.g. here it would be:

```
knownLevels.collect { case (level, name) if level == this => name }.headOption
```

And pretty similar code in `fromString`.

Reply:
This sounds good, will do. A slightly unrelated point: I feel that the name `fromString` somehow implies that it's the opposite of `toString`. What do you think about renaming it to `fromName` now that we have `name`?

Reply:
> What do you think about renaming 

It's a public method so renaming means breaking compatibility.